producer-c/producer-cloudwatch-integ/CanaryStreamCallbacks.cpp

```diff
@@ -0,0 +1,216 @@
+
+    // Set the version, self
+    pCanaryStreamCallbacks->streamCallbacks.version = STREAM_CALLBACKS_CURRENT_VERSION;
+    pCanaryStreamCallbacks->streamCallbacks.customData = (UINT64) pCanaryStreamCallbacks;
+    pCanaryStreamCallbacks->timeOfNextKeyFrame = new std::map<UINT64, UINT64>();
+    pCanaryStreamCallbacks->pStreamName = pStreamName;
+
+    pCanaryStreamCallbacks->pCwClient = cwClient;
+
+
+    dimension.SetName(pStreamName);
+    dimension.SetValue("ProducerSDK");
+
+    pCanaryStreamCallbacks->receivedAckDatum.SetMetricName("ReceivedAckLatency");
+    pCanaryStreamCallbacks->persistedAckDatum.SetMetricName("PersistedAckLatency");
+    pCanaryStreamCallbacks->streamErrorDatum.SetMetricName("StreamError");
+    pCanaryStreamCallbacks->currentFrameRateDatum.SetMetricName("FrameRate");
+    pCanaryStreamCallbacks->contentStoreAvailableSizeDatum.SetMetricName("StorageSizeAvailable");
+    pCanaryStreamCallbacks->currentViewDurationDatum.SetMetricName("CurrentViewDuration");
+    pCanaryStreamCallbacks->memoryAllocationSizeDatum.SetMetricName("MemoryAllocation");
+
+    pCanaryStreamCallbacks->receivedAckDatum.AddDimensions(dimension);
+    pCanaryStreamCallbacks->persistedAckDatum.AddDimensions(dimension);
+    pCanaryStreamCallbacks->streamErrorDatum.AddDimensions(dimension);
+    pCanaryStreamCallbacks->currentFrameRateDatum.AddDimensions(dimension);
+    pCanaryStreamCallbacks->contentStoreAvailableSizeDatum.AddDimensions(dimension);
+    pCanaryStreamCallbacks->currentViewDurationDatum.AddDimensions(dimension);
+    pCanaryStreamCallbacks->memoryAllocationSizeDatum.AddDimensions(dimension);
+    // Set callbacks
+    pCanaryStreamCallbacks->streamCallbacks.fragmentAckReceivedFn = CanaryStreamFragmentAckHandler;
+    pCanaryStreamCallbacks->streamCallbacks.streamErrorReportFn = CanaryStreamErrorReportHandler;
+    pCanaryStreamCallbacks->streamCallbacks.freeStreamCallbacksFn = CanaryStreamFreeHandler;
+
+CleanUp:
+
+    if (STATUS_FAILED(retStatus)) {
+        pCanaryStreamCallbacks = NULL;
+    }
+
+    // Set the return value if it's not NULL
+    if (ppStreamCallbacks != NULL) {
+        *ppStreamCallbacks = (PStreamCallbacks) pCanaryStreamCallbacks;
+    }
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS freeCanaryStreamCallbacks(PStreamCallbacks* ppStreamCallbacks)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PCanaryStreamCallbacks pCanaryStreamCallbacks = NULL;
+
+    CHK(ppStreamCallbacks != NULL, STATUS_NULL_ARG);
+
+    pCanaryStreamCallbacks = (PCanaryStreamCallbacks) *ppStreamCallbacks;
+
+    // Call is idempotent
+    CHK(pCanaryStreamCallbacks != NULL, retStatus);
+
+    delete(pCanaryStreamCallbacks->timeOfNextKeyFrame);
+    // Release the object
+    MEMFREE(pCanaryStreamCallbacks);
+
+    // Set the pointer to NULL
+    *ppStreamCallbacks = NULL;
+
+CleanUp:
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS CanaryStreamFreeHandler(PUINT64 customData)
```

Comment:
function names should be camelCased

Reply:
Agreed. Followed C++ SDK library naming convention. Fixed to camel case everywhere